test/libsolidity/syntaxTests/nameAndTypeResolution/042_functions_with_stucts_of_non_external_types_in_interface_2.sol

```diff
@@ -5,4 +5,5 @@ contract C {
     function f(S memory) public {}
 }
 // ----
+// TypeError: (105-113): Type is required to live outside storage.
```

Comment:
Can we make the error message a bit clearer? Something like "structs containing mappings cannot be used in memory or calldata"?

Reply:
@chriseth  Currently we can't, because `canLiveOutsideStorage()` returns only a `bool.` We could go back to `BoolResult` (When I started this PR, it was `BoolResult`, but the error message was never used, so I removed that change it.)

What do you think?

Reply:
I think it is fine to avoid the expression "required to live outside storage". This problem is only really relevant for mappings, right? We could say "Types containing mappings can only be used in storage."

Reply:
This is fine. I would prefer "Types containing (nested) mappings can only be used in storage." which leaves a hint to the user if they can't find a mapping directly in the struct.